File: bot-api/java/src/test/java/dev/robocode/tankroyale/botapi/BaseBotStateTest.java
package dev.robocode.tankroyale.botapi;

import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import java.util.List;
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

@Tag("BOT")
class BaseBotStateTest {

    static class TestBot extends BaseBot {
        TestBot() {
            super(BotInfo.builder()
                .setName("TestBot")
                .setVersion("1.0")
                .addAuthor("Author")
                .addGameType("classic")
                .build());
        }
    }

    @Test
    @Tag("TR-API-BOT-007")
    @Tag("LEGACY")
    void test_TR_API_BOT_007_base_bot_accessor_defaults() {
        var bot = new TestBot();
        
        // Metadata accessors should throw BotException when not connected
        assertThrows(BotException.class, bot::getMyId);
        assertThrows(BotException.class, bot::getVariant);
        assertThrows(BotException.class, bot::getVersion);
        
        // State-dependent accessors should throw BotException when no state is available
        assertThrows(BotException.class, bot::getEnergy);
        assertThrows(BotException.class, bot::getX);
        assertThrows(BotException.class, bot::getY);
        assertThrows(BotException.class, bot::getDirection);
        assertThrows(BotException.class, bot::getGunDirection);
        assertThrows(BotException.class, bot::getRadarDirection);
        assertThat(bot.getSpeed()).isEqualTo(0);
        assertThat(bot.getGunHeat()).isEqualTo(0);
        assertThat(bot.getBulletStates()).isEmpty();
        assertThrows(BotException.class, bot::getEvents);
        
        // Game setup accessors should throw BotException when no game setup is available
        assertThrows(BotException.class, bot::getArenaWidth);
        assertThrows(BotException.class, bot::getArenaHeight);
        assertThrows(BotException.class, bot::getGameType);
    }

    @Test
    @Tag("TR-API-BOT-008")
    @Tag("LEGACY")
    void test_TR_API_BOT_008_adjustment_flags_default_false() {
        var bot = new TestBot();
        
        assertThat(bot.isAdjustGunForBodyTurn()).isFalse();
        assertThat(bot.isAdjustRadarForBodyTurn()).isFalse();
        assertThat(bot.isAdjustRadarForGunTurn()).isFalse();
    }
}

File: bot-api/dotnet/test/src/BaseBotStateTest.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;

namespace Robocode.TankRoyale.BotApi.Tests;

[TestFixture]
[Category("BOT")]
public class BaseBotStateTest
{
    private class TestBot : BaseBot
    {
        public TestBot() : base(new BotInfo("TestBot", "1.0", new List<string> { "Author" }, null, null, null, new HashSet<string> { "classic" }, null, null, null))
        {
        }
    }

    [Test]
    [Property("ID", "TR-API-BOT-007")]
    [Category("LEGACY")]
    public void Test_TR_API_BOT_007_Base_Bot_Accessor_Defaults()
    {
        var bot = new TestBot();

        // Metadata accessors should throw BotException when not connected
        Assert.Throws<BotException>(() => _ = bot.MyId);
        Assert.Throws<BotException>(() => _ = bot.Variant);
        Assert.Throws<BotException>(() => _ = bot.Version);

        // State-dependent accessors should throw BotException when no state is available
        Assert.Throws<BotException>(() => _ = bot.Energy);
        Assert.Throws<BotException>(() => _ = bot.X);
        Assert.Throws<BotException>(() => _ = bot.Y);
        Assert.Throws<BotException>(() => _ = bot.Direction);
        Assert.Throws<BotException>(() => _ = bot.GunDirection);
        Assert.Throws<BotException>(() => _ = bot.RadarDirection);
        Assert.That(bot.Speed, Is.EqualTo(0));
        Assert.That(bot.GunHeat, Is.EqualTo(0));
        Assert.That(bot.BulletStates, Is.Empty);
        Assert.Throws<BotException>(() => _ = bot.Events);

        // Game setup accessors should throw BotException when no game setup is available
        Assert.Throws<BotException>(() => _ = bot.ArenaWidth);
        Assert.Throws<BotException>(() => _ = bot.ArenaHeight);
        Assert.Throws<BotException>(() => _ = bot.GameType);
    }

    [Test]
    [Property("ID", "TR-API-BOT-008")]
    [Category("LEGACY")]
    public void Test_TR_API_BOT_008_Adjustment_Flags_Default_False()
    {
        var bot = new TestBot();

        Assert.That(bot.AdjustGunForBodyTurn, Is.False);
        Assert.That(bot.AdjustRadarForBodyTurn, Is.False);
        Assert.That(bot.AdjustRadarForGunTurn, Is.False);
    }
}
